test/composition/declarative-shadow-dom.html

```diff
@@ -80,6 +80,17 @@
                     .equal(document.querySelector(REFERENCE_COMPOSITION).innerHTML.trim());
             });
         });
+        describe('after partial property is replaced with itself', function () {
+            it('should keep using the existing composition', function (done) { //otherwise ViewKeeper (aka workspaces) do not keep state in Firefox and Edge
+                var old  = scInclude.shadowRoot.querySelector("h2");
+                scInclude.partial = scInclude.partial;
+                setTimeout(function() {
+                    expect(old).to.be.not.null;
+                    expect(old).to.be.equal(scInclude.shadowRoot.querySelector("h2"));
```

Comment:
But that way, you are not testing "keep", you are testing "is again/still the markup". I think we should avoid flashing and in case there is a custom element that animates something, for example, we shouldn't break it.

Reply:
No, line 89 checks the equality of the DOM node, not just the equality of the node's name.

Perhaps we could check for flashing caused by detaching and attaching the same node, is that what you mean? I guess a way to do that would be to use mutation observer.

Reply:
The fact that one particular node is back there again does not mean it didn't flashed due to:
- detaching and attaching the same node,
- attaching some other nodes,
- removing some other nodes,
- removing entire composition, putting another one, then restoring this back.

Shadow DOM (composition) is a `DocumentFragment` and that's why I believe we should treat it as a whole, not just individual elements.

> I guess a way to do that would be to use mutation observer.

Great idea. :)

Reply:
But would comparing `DocumentFragment` would test against flashing? I think we can only test flashing with mutation observer, no?